### PR TITLE
16444 Pass keycloak roles to LaunchDarkly

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#16444

*Description of changes:*
- app version = 6.4.1
- added getKeycloakRoles (store getter)
- renamed local getKeycloakRoles -> fetchKeycloakRoles
- now pass Keycloak roles to LaunchDarkly
- misc cleanup

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by severinbeauvais** [comment]
Before:

image

After:

image

File: src/App.vue [hunk 1]
```diff
@@ -374,7 +375,7 @@ export default {
       try {
         // get Keycloak roles
         const jwt = this.getJWT()
-        const keycloakRoles = this.getKeycloakRoles(jwt)
```
Comment (severinbeauvais) [inline src/App.vue:377]: Renamed to avoid conflict with store getter.

File: src/App.vue [hunk 2]
```diff
@@ -678,7 +679,7 @@ export default {
           applicationDate: this.dateToYyyyMmDd(this.apiToDate(header.date)),
           legalFilings: [header.name]
         }
-      }
+      } as ApiFilingIF
```
Comment (severinbeauvais) [inline src/App.vue:682]: I did this to cast the incomplete object into the desired type.

**Review by vysakh-menon-aot — APPROVED** [review]

**Review by argush3 — APPROVED** [review]

**Review by ketaki-deodhar — APPROVED** [review]

**Review by JazzarKarim — APPROVED** [review]
LGTM!

File: src/App.vue [hunk 3]
```diff
@@ -541,8 +542,8 @@ export default {
       const email: string = userInfo.contacts[0]?.email || userInfo.email
       const firstName: string = userInfo?.firstname
       const lastName: string = userInfo?.lastname
-      // remove leading { and trailing } and tokenize string
-      const custom: any = { roles: userInfo.roles?.slice(1, -1).split(',') }
```
Comment (kialj876) [inline src/App.vue:545]: Hey Severin, would it be better to update the backend call to return the roles here? That way you wouldn't need to make this change across the other UIs and anything else using the same flow would automatically get this change in their ldarkly users

Comment (kialj876) [reply src/App.vue:545]: Or maybe you can remove calling the api for the userInfo altogether? I think all of the information is within the JWT which you are already parsing within the UI. Maybe there's another reason I'm not aware of

Comment (severinbeauvais) [reply src/App.vue:545]: The UI gets the feature flags in the `user` response from LD.

Eg, payload:
image

Eg, response:
image

I'm afraid both the UI and the API will need to "identify" the user. I don't know whether the user's attributes is retained between `user` calls. For example, if the API provides a custom object but the UI doesn't, will LD use the custom fields from the first call?

Comment (severinbeauvais) [reply src/App.vue:545]: Yes, it looks like the UI can obtain most, if not all, the user info from the JWT, so fetching the user info from the Auth API (`@me`) may not be needed.

Comment (severinbeauvais) [reply src/App.vue:545]: LMK if OK to merge or if you want some changes.

**Pull request merged**